cli/clibase/option.go

```diff
@@ -23,6 +23,10 @@ const (
 type Option struct {
 	Name        string `json:"name,omitempty"`
 	Description string `json:"description,omitempty"`
+	// Required means this value must be set by some means. It requires
+	// `ValueSource != ValueSourceNone`
+	// If `Default` is set, then `Required` is ignored.
```

Comment:
It cannot be a validation rule because the validation rule is only called on `Set()`. If the user never provides a value, and no default is set, then `Set()` is never called.

So it has to happen in the flag parsing logic.